updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+require "dependabot/dependency_change_builder"
+
+# This module contains the methods required to build a DependencyChange for
+# a single DependencyGroup.
+#
+# When included in an Operation it expects the following to be available:
+# - job: the current Dependabot::Job object
+# - dependency_snapshot: the Dependabot::DependencySnapshot of the current
+#   repo state
+# - error_handler: a Dependabot::UpdaterErrorHandler to report
+#
+module Dependabot
+  class Updater
+    module GroupUpdateCreation
+      # Returns a Dependabot::DependencyChange object that encapsulates the
+      # outcome of attempting to update every dependency iteratively which
+      # can be used for PR creation.
+      def compile_all_dependency_changes_for(group)
+        all_updated_dependencies = []
+        updated_files = []
+        group.dependencies.each do |dependency|
```

Comment:
@Nishnha I just want to call out this change so it doesn't get washed out in the move, I changed this line from

```ruby
dependencies.each do |dependency|
            next unless group.contains?(dependency)
```

In my testing, the two sets were always the same so it semantically made more sense to me to use `group.dependencies` and it also made this method agnostic of whether we just have a target group or several groups.

Reply:
Yep this would be identical. After the dependency groups are calculated (which is a side effect of calling DependencySnapshot#job_groups) we can just refer to `group.dependencies`